src/function/arithmetic/ceil.js

```diff
@@ -122,6 +128,19 @@ export const createCeil = /* #__PURE__ */ factory(name, dependencies, ({ typed,
       return x.ceil(n.toNumber())
     },
 
+    'Unit, number | BigNumber, Unit': typed.referToSelf(self => function (x, n, unit) {
+      if (typeof n !== 'number') n = n.toNumber()
```

Comment:
Can you remove the line `if (typeof n !== 'number') n = n.toNumber()` from each of the functions? We're calling `self(valueless, n)`, and that will do the checking on whether `n` is a number or BigNumber itself :)

Reply:
Got it thanks!

Reply:
Closing this comment in favor of https://github.com/josdejong/mathjs/pull/3269#discussion_r1779340372